roles/openshift_logging_fluentd/templates/fluentd.j2

```diff
@@ -66,7 +66,9 @@ spec:
           readOnly: true
         - name: filebufferstorage
           mountPath: /var/lib/fluentd
-{% if openshift_logging_mux_client_mode is defined %}
+{% if ((openshift_logging_mux_client_mode is defined and openshift_logging_mux_allow_external == true)
+   or (openshift_logging_mux_client_mode is defined and openshift_logging_mux_allow_external == false and openshift_logging_use_mux == true))
+%}
```

Comment:
the correct way to do this is pretty ugly:
```
{% if openshift_logging_mux_client_mode is defined and
     ((openshift_logging_mux_allow_external is defined and openshift_logging_mux_allow_external | bool) or
      (openshift_logging_use_mux is defined and openshift_logging_use_mux | bool)) %}
```
You have to test if the variable is defined first, then see if the boolean value is true or false.